espresso/submitter/multi_worker_queue_espresso_submitter.go

```diff
@@ -0,0 +1,1175 @@
+// transaction hash that was being fetched.
+type ErrorFetchTransactionByHashFailed struct {
+	Cause    error
+	WorkerID uint64
+	TxnHash  espresso_types.TaggedBase64
+}
+
+// Error implements error
+func (e ErrorFetchTransactionByHashFailed) Error() string {
+	return fmt.Sprintf("failed to fetch transaction by hash %s: %v", e.TxnHash.String(), e.Cause)
+}
+
+// Unwrap provides the underlying error for builtin errors checking
+func (e ErrorFetchTransactionByHashFailed) Unwrap() error {
+	return e.Cause
+}
+
+// ErrorFetchTransactionsForBlockAndNamespaceFailed is an error that is
+// returned when fetching transactions for a specific block and namespace
+// fails.
+type ErrorFetchTransactionsForBlockAndNamespaceFailed struct {
+	Cause       error
+	WorkerID    uint64
+	BlockHeight uint64
+	Namespace   uint64
+	TxnHash     espresso_types.TaggedBase64
+}
+
+// Error implements error
+func (e ErrorFetchTransactionsForBlockAndNamespaceFailed) Error() string {
+	return fmt.Sprintf("failed to fetch transactions for block %d and namespace %d, determined from transaction hash: %s: %v", e.BlockHeight, e.Namespace, e.TxnHash.String(), e.Cause)
+}
+
+// Unwrap provides the underlying error for builtin errors checking
+func (e ErrorFetchTransactionsForBlockAndNamespaceFailed) Unwrap() error {
+	return e.Cause
+}
+
+// ErrorTransactionNotFoundInBlock is an error that is returned when a
+// transaction is not found in a specific block.
+type ErrorTransactionNotFoundInBlock struct {
+	WorkerID    uint64
+	TxnHash     espresso_types.TaggedBase64
+	BlockHeight uint64
+}
+
+// Error implements error
+func (e ErrorTransactionNotFoundInBlock) Error() string {
+	return fmt.Sprintf("transaction %s not found in block %d", e.TxnHash.String(), e.BlockHeight)
+}
+
+// startWorker is the function that actually performs the underlying work of
+// the transaction included queue worker.
+//
+// It is responsible for checking if a transaction is included in Espresso,
+// and handling the response from Espresso.
+//
+// NOTE: This worker does not observe the context cancellation, as it is
+// expected to run until it is out of work.  Instead, it relies its job
+// queue being closed by the worker scheduler to indicate that no more
+// work is left for it to perform, then it will exit gracefully.
+func (w *transactionIncludedQueueWorker) startWorker(_ context.Context) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ch := make(chan TransactionIncludedJob)
+	for {
+		// Submit our worker channel to the job queue
+		w.transactionIncludedJobQueue <- ch
+		// Wait for a job to be sent to us
+		job, ok := <-ch
+		if !ok {
+			log.Info("Transaction inclusion job queue closed, exiting", "worker", w.id)
+			return
+		}
+		// Process the job
+
+		if job.attempt > 0 {
+			// Let's slow down a little
+			time.Sleep(convertToInt64WithFallback[uint, time.Duration](job.attempt, 0) * 100 * time.Millisecond)
```

Comment:
Should we make this configurable?

Reply:
Addressed in ec77acd5e4b5dba0af7c53849cf36455d0dd5e53.